reframe/core/config.py

```diff
@@ -381,16 +381,16 @@ def convert_old_config(filename, newfilename=None):
         'environments': [],
         'logging': [],
     }
-    perflogdir = None
+    perflogdir = {}
     old_systems = old_config.site_configuration['systems'].items()
     for sys_name, sys_spec in old_systems:
         sys_dict = {'name': sys_name}
 
-        # FIXME: We pick the perflogdir that we first find we use it as
-        # filelog's basedir for all systems. This is not correct since
-        # per-system customizations of perflogdir will be lost
-        if perflogdir is None:
-            perflogdir = sys_spec.pop('perflogdir', None)
+        system_perflogdir = sys_spec.pop('perflogdir', None)
+        if system_perflogdir in perflogdir:
+            perflogdir[system_perflogdir].append(sys_name)
+        else:
+            perflogdir[system_perflogdir] = [sys_name]
```

Comment:
I put `None` to have all the systems that don't define `perflogdir`. It's easier to not print the `'target_systems': target_systems` line for the systems that don't define it. Otherwise I could count how many systems are in the `perflogdir` dictionary (something like `sum([len(l) for l in system_perflogdir.values()]`) and if they are less than the number of the systems then I print the "generic" handlers

Reply:
Ok, I agree, you can keep them if you plan to use them later. It was just not clear why you did that.

Reply:
I removed `target_systems` for the systems that don't define perflogdir and kept the `None` entry.